code-samples/eventing/bookstore-sample-app/ML-sentiment-analysis/README.md

```diff
@@ -0,0 +1,311 @@
+knative function will ensure that the `setup.py` file is executed after the dependencies have been installed.
+
+
+The `setup.py` file should contain the following code for your bookstore:
+
+
+```python
+from setuptools import setup, find_packages
+from setuptools.command.install import install
+import subprocess
+
+
+class PostInstallCommand(install):
+    """Post-installation for installation mode."""
+    def run(self):
+        # Call the superclass run method
+        install.run(self)
+        # Run the command to download the TextBlob corpora
+        subprocess.call(['python', '-m', 'textblob.download_corpora', 'lite'])
+
+
+setup(
+    name="download_corpora",
+    version="1.0",
+    packages=find_packages(),
+    cmdclass={
+        'install': PostInstallCommand,
+    }
+)
+```
+
+
+
+
+### Step 5: Try to build and run your Knative Function on your local machine
+
+In knative function, there are two ways to build: using the [pack build](https://github.com/knative/func/blob/8f3f718a5a036aa6b6eaa9f70c03aeea740015b9/docs/reference/func_build.md?plain=1#L46) or using the [source-to-image (s2i) build](https://github.com/knative/func/blob/4f48549c8ad4dad34bf750db243d81d503f0090f/docs/reference/func_build.md?plain=1#L43). 
+
+Currently. only the **s2i** build is supported if you need to run setup.py. When building with s2i, the `setup.py` file will be executed automatically after the dependencies have been installed.
+
+Before we get started, configure the container registry to push the image to the container registry. You can use the following command to configure the container registry:
+
+```bash
+export FUNC_REGISTRY=<your-container-registry>
+```
+
+In this case, we will use the s2i build by adding the flag `-b=s2i`, and `-v` to see the verbose output.
+
+```bash
+func build -b=s2i -v
+```
+
+When the build is complete, you will see the following output:
+
+```bash
+🙌 Function built: <Your container registry username>/sentiment-analysis-app:latest
+```
+
+This command will build the function and push the image to the container registry. After the build is complete, you can run the function using the following command:
+
+```bash
+func run
+```
+In the future, you can skip the step of func build, because func run will automatically build the function for you.
+
+You will see the following output if the function is running successfully:
+
+```
+function up-to-date. Force rebuild with --build
+Running on host port 8080
+---> Running application from script (app.sh) ...
+````
+
+Now you can test the function by sending a request to the function using the following command:
+
+```bash
+curl -X POST http://localhost:8080/ \
+-H "ce-id: 12345" \
+-H "ce-source: my-local" \
+-H "ce-type: sentiment-analysis-request" \
+-H "ce-specversion: 1.0" \
+-H "Content-Type: application/json" \
+-d '{"input": "I love Knative so much!"}'
+```
```

Comment:
would `func invoke` work here?

Reply:
It should be working properly right now. @Cali0707